ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java

```diff
@@ -1167,6 +1180,75 @@ public void onAppendEntriesReply(LogAppender appender, RaftProtos.AppendEntriesR
     readIndexHeartbeats.onAppendEntriesReply(appender, reply, this::hasMajority);
   }
 
+  @Override
+  public void onTransferLeadership(RaftPeerId transferee) {
```

Comment:
We add the new logic to `TransferLeadership`.  The new logic is to sends a TimeoutNow request to the target server.

Reply:
The current `LeaderState#sendStartLeaderElectionToHigherPriorityPeer()` is actually `sendTimeoutNow()`.
However, I think it should be refactored in another PR.